src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -820,14 +821,14 @@ namespace winrt::TerminalApp::implementation
             std::filesystem::path cwd{ cwdString };
             cwd /= settings.StartingDirectory().c_str();
 
-            auto conhostConn = TerminalConnection::ConptyConnection(
-                settings.Commandline(),
-                winrt::hstring{ cwd.c_str() },
-                settings.StartingTitle(),
-                envMap.GetView(),
-                settings.InitialRows(),
-                settings.InitialCols(),
-                winrt::guid());
+            auto conhostConn = TerminalConnection::ConptyConnection();
+            conhostConn.Initialize(TerminalConnection::ConptyConnection::CreateSettings(settings.Commandline(),
+                                                                                        winrt::hstring{ cwd.c_str() },
```

Comment:
```suggestion
                                                                                        winrt::hstring{ cwd.wstring() },
```

does this work?

Reply:
(In general, if you find yourself going through c_str when you have a smart string type, it probably costs you another `strlen` that you could have avoided.)